tuta-sdk/rust/src/util/test_utils.rs

```diff
@@ -1,10 +1,105 @@
 //! General purpose functions for testing various objects
 
 use crate::crypto::randomizer_facade::test_util::make_thread_rng_facade;
+use crate::custom_id::CustomId;
+use crate::element_value::{ElementValue, ParsedEntity};
+use crate::entities::Entity;
+use crate::generated_id::GeneratedId;
+use crate::IdTuple;
+use crate::instance_mapper::InstanceMapper;
+use crate::metamodel::{AssociationType, Cardinality, ElementType, ValueType};
+use crate::type_model_provider::{init_type_model_provider, TypeModelProvider};
 
 /// Generates a URL-safe random string of length `Size`.
+#[must_use]
 pub fn generate_random_string<const SIZE: usize>() -> String {
     use base64::engine::Engine;
     let random_bytes: [u8; SIZE] = make_thread_rng_facade().generate_random_array();
     base64::engine::general_purpose::URL_SAFE.encode(random_bytes)
 }
+
+/// Generate a test entity.
+///
+/// The values will be set to these defaults:
+/// * All ZeroOrOne values will be null
+/// * All Any values will be empty
+/// * All One values will use default values, or random values if an ID type
+///
+/// # Examples
+///
+/// ```ignore
+/// use crate::entities::tutanota::Mail;
+/// use crate::util::test_utils::create_test_entity;
+///
+/// let mail = Mail {
+///     phishingStatus: 1337, // 😎
+///     ..create_test_entity()
+/// };
+///
+/// assert_eq!(1337, mail.phishingStatus);
+/// ```
+#[must_use]
+pub fn create_test_entity<'a, T: Entity + serde::Deserialize<'a>>() -> T {
+    let provider = init_type_model_provider();
+    let type_ref = T::type_ref();
+    let mapper = InstanceMapper::new();
+    let entity = create_test_entity_dict(&provider, type_ref.app, type_ref.type_);
```

Comment:
this is a wonderful workaround! I applaud your ingenuity.

A more complicated solution would involve a macro that sets some of the fields and then overwrites those we specify but guess what! we don't need it for now!